spec/index.adoc

```diff
@@ -530,7 +530,11 @@ In order to call a canister, the user makes a POST request to `/api/v2/canister/
 * `method_name` (`text`): Name of the canister method to call
 * `arg` (`blob`): Argument to pass to the canister method
 
-The HTTP response to this request has an empty body and HTTP status 202, or a HTTP error (4xx or 5xx). Paranoid agents should not trust this response, and use <<http-read-state,`read_state`>> to determine the status of the call.
+The HTTP response to this request can have the following responses:
+* 202 HTTP status with empty body. Implying the request was accepted by the IC for further processing. Users should use <<http-read-state,`read_state`>> to determine the status of the call.
+* 200 HTTP status with non-empty body. Implying an execution pre-processing error occurred. The body of the response contains more information about the user error. This return type
+is used when the error can be handled by the canister owner (e.g. when a canister is out of cycles) 
```

Comment:
The previous sentence talks about "user" error (sounds like it's triggered by the request sent by the user) while the last sentence talks about "canister owner" to fix the error (although canister running out of cycles needn't be only fixed by the controller). Could you please clarify what types of error are expected there? @rumenov

Reply:
I am not an expert [on execution errors](https://github.com/dfinity/ic/blob/master/rs/http_endpoints/public/src/common.rs#L80) but I believe most of the issues here should be returned as OK 200

@dsarlis any suggestions how to improve the wording here ?

Reply:
If we do this, wouldn't then it mean that 4xx and 5xx can be returned only when the request is malformed (or something along those lines) or the replica is unavailable respectively? And therefore, any IC specific error would be returned via 200 with non-empty body containing the details of the error?

Reply:
Oh wait, too early in the morning, I misread the bullet points. I think we can simply say something like:
```
200 HTTP status with non-empty body. Implying an execution pre-processing error occurred. The body of the response contains more information about the IC specific error encountered.
```
and completely drop the last part. Some of the errors could be handled only by the canister owner or they could be handled by the client too, so trying to be more specific doesn't help much.

Reply:
Also, I think we should specify the body in the case where an error is returned. I'd expect it would be quite similar to the error case for a query call. So, something like:

```
The body is a CBOR map with the following fields:
* `status` (`text`): `rejected`
* `reject_code` (`nat`): The reject code (see <<reject-codes>>).
* `reject_message` (`text`): a textual diagnostic message.
* `error_code` (text): an optional implementation-specific textual error code (see <<error-codes>>).
```
(maybe we can skip `status` in this case though as it would only be non-empty in the error case anyway).

Reply:
@dsarlis can you please take another look ?

Reply:
Lgtm!